### PR TITLE
Support ES6 files and add karma test framework

### PR DESCRIPTION
**Pull request by kylezeeuwen**

### CODE REVIEW

File: theSrc/scripts/PlotColors.es6.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// This class assigns colors to the plot data
+// If group array is supplied, it will couple the array with the color wheel
+// If no group array is supplied, colors are rotated through the color wheel
+class PlotColors {
+  constructor(plotData) {
+    this.plotData = plotData;
+    this.plotData.legendGroups = [];
+
+    this.groupToColorMap = {};
+
```
Comment (kylezeeuwen) [inline theSrc/scripts/PlotColors.es6.js]: i rerwrote this loop. In future I would like to pull out the side effect part, but for now just left it as a note to refactor

File: theSrc/scripts/PlotData.coffee [hunk 2]
```diff
@@ -23,19 +23,17 @@ class PlotData
     @origY = @Y.slice(0)
     @normX = @X.slice(0)
     @normY = @Y.slice(0)
```
Comment (kylezeeuwen) [inline theSrc/scripts/PlotData.coffee]: I changed (everywhere) from Utils.get() to Utils

File: theSrc/scripts/PlotData.coffee [hunk 3]
```diff
@@ -23,19 +23,17 @@ class PlotData
...
-    @normZ = @Z.slice() if Utils.get().isArrOfNums(@Z) and @Z.length == @X.length
+    @normZ = @Z.slice() if Utils.isArrOfNums(@Z) and @Z.length == @X.length
     @outsidePlotPtsId = []
     @legendPts = []
     @outsidePlotCondensedPts = []
     @legendBubbles = []
     @legendBubblesLab = []
```
Comment (kylezeeuwen) [inline theSrc/scripts/PlotData.coffee]: cIndex is no longer needed and is not used anywhere (except old PlotColors code)

File: theSrc/scripts/PlotData.coffee [hunk 4]
```diff
@@ -445,7 +444,7 @@ class PlotData
     }
     @outsidePlotPtsId.push id
     @normalizeData()
-    @getPtsAndLabs()
+    @getPtsAndLabs('addElemFromLegend')
```
Comment (poliu2s) [inline theSrc/scripts/PlotData.coffee]: Is this a way to comment on how the code enters the function? Or was this part of some previous debug code?

File: theSrc/scripts/PlotData.coffee [hunk 5]
```diff
@@ -36,7 +36,6 @@ class PlotData
       @normalizeZData() if Utils.isArrOfNums(@Z)
       @plotColors = new PlotColors(@)
       @labelNew = new PlotLabel(@label, @labelAlt, @viewBoxDim.labelLogoScale)
-      @getPtsAndLabs()
```
Comment (poliu2s) [inline theSrc/scripts/PlotData.coffee]: Good pick up on the extra call 👍

File: theSrc/scripts/PlotLabel.es6.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+ */
+
+// PlotLabel class
+// The primary purpose of this class is to support images as the labels and
+// to parse them apart from the regular text labels
+
+class PlotLabel {
+
+  constructor(givenLabelArray, labelAlt, logoScale) {
+    this._makeLabPromise = this._makeLabPromise.bind(this);
```
Comment (poliu2s) [inline theSrc/scripts/PlotLabel.es6.js]: With so many this's, makes you miss coffee a bit 😄

File: theSrc/scripts/PlotData.coffee (hunk 3 above)
Comment (poliu2s) [inline theSrc/scripts/PlotData.coffee]: 👍

File: theSrc/scripts/PlotColors.es6.js (hunk 1 above)
Comment (poliu2s) [inline theSrc/scripts/PlotColors.es6.js]: 👍

File: theSrc/scripts/PlotData.coffee (hunk 2 above)
Comment (poliu2s) [inline theSrc/scripts/PlotData.coffee]: 👍

File: theSrc/scripts/utils/Utils.spec.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  describe('isNum():', function () {
+    const tests = [
+      { input: 4, expected: true },
+      { input: '4', expected: false },
+      { input: ['4'], expected: false },
+      { input: undefined, expected: false },
+      { input: null, expected: false },
+    ];
+
+    _(tests).each((t) => {
```
Comment (poliu2s) [inline theSrc/scripts/utils/Utils.spec.js]: _(variable) --> I haven't seen this syntax before, is this for const collections in ES6?

File: theSrc/scripts/PlotData.coffee (hunk 4 above)
Comment (kylezeeuwen) [inline theSrc/scripts/PlotData.coffee]: no that is left over debug code - I was tracking down why it was being called too many times -
 will delete

File: theSrc/scripts/PlotLabel.es6.js (hunk 6 above)
Comment (kylezeeuwen) [inline theSrc/scripts/PlotLabel.es6.js]: ya a bit ... but not much

**Pull request merged**